src/pages/upload/index.tsx

```diff
@@ -74,88 +86,206 @@ const fileUploadReducer = (state: any, action: any) => {
   }
 }
 
-const Upload: React.FC<{instructors: Instructor[]}> = ({instructors}) => {
-  // TODO: rename this reducer variables
-  const [state, dispatch] = React.useReducer(fileUploadReducer, {files: []})
-  const uploaderRef = React.useRef(null)
+const UploadWrapper = ({instructors}: {instructors: Instructor[]}) => {
+  const initialValues: FormProps = {lessons: []}
+
+  return (
+    <Formik
```

Comment:
Wrapping the whole thing in Formik to make form state management easy and for handling form submission (via new API endpoint).